src/player_display.cpp

```diff
@@ -488,51 +598,58 @@ Strength - 4;    Dexterity - 4;    Intelligence - 4;    Perception - 4" ) );
     bionics_win_size_y--;
     skill_win_size_y--;
     effect_win_size_y--;
+}
 
+input_context player_display_info::draw_header(player& player)
+{
     // Print name and header
     // Post-humanity trumps your pre-Cataclysm life.
-    if( crossed_threshold() ) {
+    if( player.crossed_threshold() ) {
         std::string race;
-        for( auto &mut : my_mutations ) {
-            const auto &mdata = mut.first.obj();
+        for( const auto &mut : player.get_mutations() ) {
+            const auto &mdata = mut.obj();
             if( mdata.threshold ) {
                 race = mdata.name();
                 break;
             }
         }
         //~ player info window: 1s - name, 2s - gender, 3s - Prof or Mutation name
-        mvwprintw( w_tip, 0, 0, _( "%1$s | %2$s | %3$s" ), name.c_str(),
-                   male ? _( "Male" ) : _( "Female" ), race.c_str() );
-    } else if( prof == NULL || prof == prof->generic() ) {
+        mvwprintw( w_tip, 0, 0, _( "%1$s | %2$s | %3$s" ), player.name.c_str(),
+                   player.male ? _( "Male" ) : _( "Female" ), race.c_str() );
+    } else if( player.prof == NULL || player.prof == player.prof->generic() ) {
         // Regular person. Nothing interesting.
         //~ player info window: 1s - name, 2s - gender, '|' - field separator.
-        mvwprintw( w_tip, 0, 0, _( "%1$s | %2$s" ), name.c_str(),
-                   male ? _( "Male" ) : _( "Female" ) );
+        mvwprintw( w_tip, 0, 0, _( "%1$s | %2$s" ), player.name.c_str(),
+                   player.male ? _( "Male" ) : _( "Female" ) );
     } else {
-        mvwprintw( w_tip, 0, 0, _( "%1$s | %2$s | %3$s" ), name.c_str(),
-                   male ? _( "Male" ) : _( "Female" ), prof->gender_appropriate_name( male ).c_str() );
+        mvwprintw( w_tip, 0, 0, _( "%1$s | %2$s | %3$s" ), player.name.c_str(),
+                   player.male ? _( "Male" ) : _( "Female" ), player.prof->gender_appropriate_name( player.male ).c_str() );
     }
 
     input_context ctxt( "PLAYER_INFO" );
     ctxt.register_updown();
+    ctxt.register_action( "PREV_TAB");
     ctxt.register_action( "NEXT_TAB", _( "Cycle to next category" ) );
     ctxt.register_action( "QUIT" );
     ctxt.register_action( "CONFIRM", _( "Toggle skill training" ) );
     ctxt.register_action( "HELP_KEYBINDINGS" );
-    std::string action;
 
     std::string help_msg = string_format( _( "Press %s for help." ),
                                           ctxt.get_desc( "HELP_KEYBINDINGS" ).c_str() );
     mvwprintz( w_tip, 0, FULL_SCREEN_WIDTH - utf8_width( help_msg ), c_light_red, help_msg.c_str() );
     help_msg.clear();
```

Comment:
This doesn't need clearing, the variable is destroyed at the function end anyway. And the variable is not needed anyway, `mvwprintz` above does string formatting itself: `mvwprintz( w_tip, ..., _( "Press %s" ), get_desc() );`